crates/polars-stream/src/nodes/ordered_union.rs

```diff
@@ -0,0 +1,84 @@
+use parking_lot::Mutex;
+
+use super::compute_node_prelude::*;
+
+/// A node that first passes through all data from the first input, then the
+/// second input, etc.
+pub struct OrderedUnionNode {
+    cur_input_idx: usize,
+    max_morsel_seq_sent: Mutex<MorselSeq>,
```

Comment:
Why don't we use atomics for this?

Reply:
Then you can't use `MorselSeq`, and it's really not performance sensitive at all, it only occurs once per execution phase.